### PR TITLE
exif: tentative fix for MacOS build issue.

### PR DESCRIPTION
**Pull request by TurboGit**

### CODE REVIEW

**Comment by TurboGit** [comment]
@kmilos @victoryforce : Is that looks correct to you?

**Comment by victoryforce** [comment]
> @kmilos @victoryforce : Is that looks correct to you?

The fix looks correct. The problem that the LLVM compiler sees is that the expression fragments like `history->key()` or `history->value()` returns a temporary `std::string`. The compiler warns that it may be destoyed before the next statement. So, yes, the fix is right.

File: src/common/exif.cc [hunk 1]
```diff
@@ -2600,9 +2600,10 @@ static GList *read_history_v1(const std::string &xmpPacket, const char *filename
 
     if(blendop_params && blendop_params_iter != blendop_params.node().children().end())
     {
-      current_entry->blendop_params = dt_exif_xmp_decode(blendop_params_iter->child_value(),
-                                                         strlen(blendop_params_iter->child_value()),
-                                                         &current_entry->blendop_params_len);
+      current_entry->blendop_params = dt_exif_xmp_decode
+        (blendop_params_iter->child_value(),
+         strlen(blendop_params_iter->child_value()),
+         &current_entry->blendop_params_len);
```
[the remark this answers is not in the document]
Comment (TurboGit) [reply src/common/exif.cc:2606]: Yes, will correct.

File: src/common/exif.cc [hunk 2]
```diff
@@ -2725,9 +2732,11 @@ static GList *read_history_v2(Exiv2::XmpData &xmpData, const char *filename)
...
       else if(g_str_has_prefix(key_iter, "darktable:blendop_params"))
       {
-        current_entry->blendop_params = dt_exif_xmp_decode(history->value().toString().c_str(),
-                                                           history->value().size(),
-                                                           &current_entry->blendop_params_len);
+        std::string value_item = history->value().toString();
+        current_entry->blendop_params = dt_exif_xmp_decode
+          (value_item.c_str(),
+           history->value().size(),
+           &current_entry->blendop_params_len);
```
[the remark this answers is not in the document]
Comment (TurboGit) [reply src/common/exif.cc:2739]: Likewise.

**Pull request merged**